File: src/main/java/com/ahold/technl/sandbox/service/DeliveryService.java
package com.ahold.technl.sandbox.service;

import com.ahold.technl.sandbox.dto.BusinessSummary;
import com.ahold.technl.sandbox.dto.DeliveryIdRecord;
import com.ahold.technl.sandbox.dto.DeliveryInvoiceRecord;
import com.ahold.technl.sandbox.dto.DeliveryRecord;

import java.util.List;

public interface DeliveryService {

    DeliveryRecord createDelivery(DeliveryRecord deliveryRecord);

    List<DeliveryInvoiceRecord> sendInvoice(DeliveryIdRecord deliveryIdRecord);

    List<DeliveryRecord> findAll();

    BusinessSummary getBusinessSummary();
}

File: src/main/kotlin/com/ahold/technl/sandbox/service/DeliveryService.kt
package com.ahold.technl.sandbox.service

import com.ahold.technl.sandbox.dto.*


interface DeliveryService {
    fun createDelivery(request: DeliveryRequest): DeliveryResponse

    fun sendInvoice(deliveryIdRecord: DeliveryIdRecord): MutableList<DeliveryInvoiceRecord?>?

    fun getBusinessSummary(): BusinessSummary
}